oci/oci.go

```diff
@@ -514,20 +514,34 @@ func (r *Runtime) StopContainer(c *Container, timeout int64) error {
 		timeout = 10
 	}
 	done := make(chan struct{})
+	// we could potentially re-use "done" channel to exit the loop on timeout
+	// but we use another channel "chControl" so that we won't never incur in the
+	// case the "done" channel is closed in the "default" select case and we also
+	// reach the timeout in the select below. If that happens we could raise
+	// a panic closing a closed channel so better be safe and use another new
+	// channel just to control the loop.
+	chControl := make(chan struct{})
 	go func() {
 		for {
-			// Check if the process is still around
-			err := unix.Kill(c.state.Pid, 0)
-			if err == syscall.ESRCH {
-				close(done)
-				break
+			select {
+			case <-chControl:
+				return
+			default:
+				// Check if the process is still around
+				err := unix.Kill(c.state.Pid, 0)
+				if err == syscall.ESRCH {
+					close(done)
+					return
+				}
+				time.Sleep(100 * time.Millisecond)
 			}
```

Comment:
we can't get stuck here at all because when the timeout is hit at line 544, we close the `chControl` (`close(chcontrol)`). After that's closed the `select` at line 527 will enter `case <-chControl:` and do a `return` exiting the loop and goroutine. That's the flow, can't see any chance to get stuck in the loop :smile: